### ISSUE TITLE
Password input in config

### ISSUE THREAD

**Issue by arist0v** [WebThingsIO/gateway#2918]
It would be usefull to have the possibility to have a password field for the addon configuration instead of a text field.

**Comment by benfrancis** [comment]
The JSON Schema Validation specification suggests using [`writeOnly`](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.9.4) for a password string.

An alternative could be to use `"format": "password"`, which isn't defined in the specification but custom format attributes are allowed. I've heard that [`format`](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7) might be removed in future versions of JSON Schema though.

**Comment by flatsiedatsie** [comment]
How would that work in practise? Something like this?

```
    "schema": {
      "properties": {
        "Your password": {
          "description": "Provide your password.",
          "type": "string",
          "writeOnly":true
        },
      }
    }
```

**Comment by arist0v** [comment]
If it mean that there is no "visualization" of the password it will be great!

**Comment by benfrancis** [comment]
> How would that work in practise? Something like this?

I believe so, yes. It seems like a bit of a hack to me to be honest, but given that's what's recommended in the specification I think it would be OK.

**Comment by flatsiedatsie** [comment]
Yeah I agree. I've implemented `"type": "password",` in Candle's app store for now. If the Gateway settles on a different route I'll add support for that too.

**Comment by benfrancis** [comment]
> I've implemented "type": "password", in Candle's app store for now.

Does using a non-standard JSON Schema type not trip up the parser? Have you tested how the gateway reacts if it comes across a type it doesn't recognise? FWIW `writeOnly` may be safer in that respect because it at least uses standard semantics.

**Comment by flatsiedatsie** [comment]
You were right, it tripped up the system parser. I've changed it to detect the `writeOnly` attribute instead.
